### PR TITLE
New structure to store form data into DB

### PR DESCRIPTION
**Pull request by ayush2k**
`DCO 1.1 Signed-off-by: {{Ayush Gupta} <{{ayush2k@gmail.com}}>`

### CODE REVIEW

**Comment by ayush2k** [comment]
@Snugug @scottnath 
It fails one test

File: lib/utils.js [hunk 1]
```diff
@@ -127,10 +127,48 @@ const config = values => {
   return result;
 };
 
+// Structures input from form
```
Comment (scottnath) [inline lib/utils.js]: Please add jsdoc code to the top of this. Should include a description `@param` and `@returns`

File: lib/utils.js [hunk 2]
```diff
@@ -127,10 +127,48 @@ const config = values => {
...
+
+      // Checks if multiple instances
+      if (inputs.length >= 3) {
+        const index = inputs[2];
+        if (!data.hasOwnProperty(inputName)) {
+          data[inputName] = [];
+        }
+        if (!data[inputName][index]) {
+          // Creates an array for multiple instances
+          data[inputName][index] = {};
```
Comment (scottnath) [inline lib/utils.js]: You're creating an object here, not an array

File: tests/utils.js [hunk 3]
```diff
@@ -122,3 +122,35 @@ test('Config', t => {
...
+
+test('Request Format', t => {
+  const input = {
+    'sunset-date': '',
+    'sunset-time': '',
+    'sunrise-date': '2016-06-23',
+    'sunrise-time': '00:00',
+    'service-name--text': 'bar',
+    'service-email--email--0': 'foo@test.com',
+    'service-email--email--1': 'bar@test.com',
```
Comment (scottnath) [inline tests/utils.js]: Please add what would be the output of a set of checkboxes or a multi-select, which should be an array.

File: tests/utils.js [hunk 4]
```diff
@@ -122,3 +122,35 @@ test('Config', t => {
...
 });
+
+test('Request Format', t => {
+  const input = {
+    'sunset-date': '',
+    'sunset-time': '',
+    'sunrise-date': '2016-06-23',
+    'sunrise-time': '00:00',
+    'service-name--text': 'bar',
+    'service-email--email--0': 'foo@test.com',
```
Comment (scottnath) [inline tests/utils.js]: Do another couple of these, making some of them blank.

File: lib/utils.js [hunk 5]
```diff
@@ -127,7 +127,14 @@ const config = values => {
   return result;
 };
 
-// Structures input from form
+/*
+ * Formats raw data from form to object
+ *
+ * @param {object} body - Raw data object
```
Comment (scottnath) [inline lib/utils.js]: this doesn't actually help much. Please use the [`typedef` descriptor](http://usejsdoc.org/tags-typedef.html) to break down what is in both the param and returns.

[Here is an example usage in our code](https://github.com/punchcard-cms/punchcard/blob/64921680ac6bd635725bc09908ea6a292dded73c/lib/utils.js#L63)

```
/*
 * @typedef FormattedDate
 * @type object
 *
 * @property {string} date - The date, formatted YYYY-MM-DD (e.g. 2016-05-25)
 * @property {string} time - The time, formatted HH:mm (e.g. 13:01)
 * @property {string} zone - The time zone, formatted Z (e.g. America/New_York)
 */

/*
 * Formats ISO date in to requisite components
 *
 * @param {string} date - ISO UTC formatted date
 * @param {string} zone - Timezone to retrieve date in
 *
 * @returns {FormattedDate} - The individually formatted date components
 */
```

**Pull request merged**